Mage.Sets/src/mage/cards/d/DesertWarfare.java

```diff
@@ -0,0 +1,225 @@
+                        || (zEvent.getFromZone() != Zone.LIBRARY && zEvent.getFromZone() != Zone.HAND)
+                        || !filterCard.match(card, getControllerId(), this, game)) {
+                    return false;
+                }
+                break;
+            default:
+                return false;
+        }
+        this.getEffects().setTargetPointer(new FixedTarget(event.getTargetId(), game));
+        return true;
+    }
+}
+
+class DesertWarfareReturnEffect extends OneShotEffect {
+
+    DesertWarfareReturnEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "put that card onto the battlefield under your control";
+    }
+
+    private DesertWarfareReturnEffect(final DesertWarfareReturnEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public DesertWarfareReturnEffect copy() {
+        return new DesertWarfareReturnEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Card card = game.getCard(getTargetPointer().getFirst(game, source));
+        if (controller != null && card != null && game.getState().getZone(card.getId()) == Zone.GRAVEYARD) {
+            controller.moveCards(card, Zone.BATTLEFIELD, source, game, false, false, false, null);
+            return true;
+        }
+        return false;
+    }
+}
+
+class DesertWarfareCreateTokensEffect extends OneShotEffect {
+
+    DesertWarfareCreateTokensEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "At the beginning of combat on your turn, if you control five or more Deserts, create that many 1/1 red, green, and white Sand Warrior creature tokens. They gain haste.";
+    }
+
+    private DesertWarfareCreateTokensEffect(final DesertWarfareCreateTokensEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public DesertWarfareCreateTokensEffect copy() {
+        return new DesertWarfareCreateTokensEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        if (source.getControllerId() == null) {
+            return false;
+        }
+        int deserts = DesertsYouControlCount.instance.calculate(game, source, this);
+        Token token = new SandWarriorToken();
+        token.putOntoBattlefield(deserts, game, source);
+        game.addEffect(new GainAbilityTargetEffect(HasteAbility.getInstance(), Duration.Custom)
+                .setTargetPointer(new FixedTargets(token, game)), source);
+        return true;
+    }
+}
+
+enum DesertsYouControlCount implements DynamicValue {
+    instance;
+
+    private static final Hint hint = new ValueHint("Deserts you control", instance);
+
+    @Override
+    public int calculate(Game game, Ability sourceAbility, Effect effect) {
+        return new PermanentsOnBattlefieldCount(new FilterControlledPermanent(SubType.DESERT)).calculate(game, sourceAbility, null);
```

Comment:
Don't wrap this in a custom class. Declare it as `private static final DynamicValue xValue` and likewise for the hint